derive-macros/src/lib.rs

```diff
@@ -84,8 +89,8 @@ fn gen_schema_fields(data: &Data) -> TokenStream {
                 });
                 if have_schema_null {
                     if let Some(first_ident) = type_path.path.segments.first().map(|seg| &seg.ident) {
-                        if first_ident != "HashMap" && first_ident != "Vec" {
-                            panic!("Can only use schema_container_values_null on HashMap or Vec fields, not {first_ident:?}");
+                        if first_ident != "HashMap" {
+                            panic!("Can only use drop_null_container_values on HashMap fields, not {first_ident:?}");
```

Comment:
IIRC `panic` technically works, because it just kills the macro invocation and produces a compiler error... but IIRC there's a better way, emitting an error to the output token stream so the span info can give a more precise error message?

Reply:
Yeah, we could use [`to_compile_error()`](https://docs.rs/syn/latest/syn/struct.Error.html)

I've created https://github.com/delta-incubator/delta-kernel-rs/issues/348 to convert to this

Reply:
TIL!